contracts/VaultV3.vy

```diff
@@ -278,16 +270,21 @@ PERMIT_TYPE_HASH: constant(bytes32) = keccak256("Permit(address owner,address sp
 
 # Constructor
 @external
-def __init__(
-    asset: ERC20, 
+def __init__():
+    # Set `asset` so it cannot be re-initialized.
+    self.asset = self
```

Comment:
the idea here is that the implementation should never be a vault itself right? 
 
 originally was thinking it'd be better to use a different burner address (e.g. `0xFFF`) , but upon further thought i think there's no issues that come from `self`, because doing so would effectively break the vault and prevent it from ever having a vault of its own.

Reply:
asset is used as the value to check during initialization to make sure it can't be re-initialized 

So just setting its to a non-zero value in the constructor stops the 'original' from being initialized at any point. Essentially making it worthless and dust a dummy version to use only for cloning.